src/main/java/com/gymrats/service/CalculatorService.java

```diff
@@ -28,7 +28,10 @@ public BmrNotLoggedResponse calculateBmr(BmrNotLoggedRequest requestBmr){
         float height = requestBmr.height();
         float bmr = 0;
 
-        if(sex.equals(GenderEnum.WOMAN)){
+        if(weight == 0 || weight < 0 || height == 0 || height < 0  || userAge == 0 || userAge < 0){
+            throw new IllegalArgumentException();
```

Comment:
no more detailed information
https://www.geeksforgeeks.org/spring-mvc-regular-expression-validation/ maybe like that?